### PR TITLE
:NORMAL: feature/style_guide_update (Big)

### PR DESCRIPTION
**Pull request by AlexJMcD**
I have made some changes to the style guide to allow more control over what you can display.

- ColorsViewHolder has been edited as before the central colour would only be highlighted if getMiddlePosition(list) returned a value which it would only do on lists with an odd number of elements, it will now also do this on lists with an even number of elements.
- Originally Buttons, checkboxes, switchs, text, and radiobutton styles were all gathered and populated in one big function, these have now been separated into individual public functions so that they can be called from a class that inherits the StyleGuideFragment.
- These functions allow for some customisation, you can pass the function a list of prefixes to ensure that it picks up your styles rather than having to tailor your style names to fit the functions. You can also tell the component functions whether or not to use material components, as styles designed specifically for material components may not work in the same way on app compat components.
- App icons have been removed as these are not needed and were causing conflicts in apps where the library was being used.

### CODE REVIEW

**Comment by AlexJMcD** [comment]
Screenshot_20230824_120936
Screenshot_20230824_123718
Screenshot_20230824_123821

**Review by Swaini — APPROVED** [review]

File: library/src/main/java/com/cube/styleguide/adapter/viewholder/ColorItemViewHolder.kt [hunk 1]
```diff
@@ -25,7 +25,9 @@ class ColorItemViewHolder(private val binding: ColorBannerViewBinding) : Recycle
 		list?.forEachIndexed { index, pair ->
 			val colorItemView = ColorItemViewBinding.inflate(LayoutInflater.from(itemView.context))
 			colorItemView.root.layoutParams = params
-			populateView(colorItemView, getMiddlePosition(list) == index, pair)
+            // If the colour list size is even it won't have a middle position, so instead we will take the index that is half of the size.
+            val isMiddle = if (list.size%2 == 0) list.size/2 else getMiddlePosition(list)
```
Comment (JR-Mitchell) [inline library/src/main/java/com/cube/styleguide/adapter/viewholder/ColorItemViewHolder.kt]: It looks like the indents have changed here - can you fix this?

Comment (AlexJMcD) [reply library/src/main/java/com/cube/styleguide/adapter/viewholder/ColorItemViewHolder.kt]: Done :)

**Review by JR-Mitchell — APPROVED** [review]
LGTM

**Pull request merged**